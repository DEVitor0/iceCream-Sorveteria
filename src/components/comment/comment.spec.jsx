import React from 'react';
import { render, screen } from '@testing-library/react';
import Comment from './index';

describe('Comment Component', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Comment />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render the opinion text correctly', () => {
    render(<Comment />);
    expect(
      screen.getByText(/Entrega rápida e funcionários simpáticos/),
    ).toBeInTheDocument();
    expect(screen.getByText(/O sorvete chegou frio e/)).toBeInTheDocument();
    expect(screen.getByText(/muito delicioso!/)).toBeInTheDocument();
  });

  it('should render the author name', () => {
    render(<Comment />);
    expect(screen.getByText(/Eva Oliveira/)).toBeInTheDocument();
  });

  it('should have the correct class names', () => {
    const { container } = render(<Comment />);
    const opinionElement = container.querySelector('.comment-block__opinion');
    const authorElement = container.querySelector(
      '.comment-block__opinion__author',
    );
    expect(opinionElement).toBeInTheDocument();
    expect(authorElement).toBeInTheDocument();
  });
});
