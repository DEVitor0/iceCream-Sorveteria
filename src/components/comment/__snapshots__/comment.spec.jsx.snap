// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="comment-block"
  >
    <p
      class="comment-block__opinion"
    >
      "Entrega rápida e funcionários simpáticos. 
      <br />
      O sorvete chegou frio e 
      <br />
      muito delicioso!
      <span
        class="comment-block__opinion__author"
      >
        Eva Oliveira
      </span>
    </p>
  </div>
</DocumentFragment>
`;
